### PR TITLE
WIP: Add file questions

### PR DESCRIPTION
**Pull request by fkm**
There are some things missing:

* [x] Tests
* [ ] Validation
* [ ] Fake answer.value if undefined
* [ ] Better mock for [GenericScalar](https://github.com/graphql-python/graphene/blob/master/graphene/types/generic.py)

### CODE REVIEW

File: addon/components/cf-field/input/file.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    async save({ target }) {
+      const file = target.files[0];
+
+      if (!file) {
+        return;
+      }
+
+      const saved = await this.onSave(file.name);
+
+      const data = new FormData();
```
Comment (czosel) [inline addon/components/cf-field/input/file.js]: Just to be sure: This works in IE11, right? ([caniuse](https://caniuse.com/#feat=xhr2) just mentions limited support for JSON content types)

Comment (fkm) [reply addon/components/cf-field/input/file.js]: This shouldn't be a problem as we append a file. But I'll check anyway.

**Comment by fkm** [comment]
There's a ticket for the missing dot:
https://github.com/projectcaluma/caluma/issues/387

I will look for a solution to the missing metadata with @open-dynaMIX tomorrow.

File: addon/components/cf-field/input/file.js (hunk 1 above)
Comment (fkm) [reply addon/components/cf-field/input/file.js]: I haven't gotten to test the FormData. IE 11 breaks long before that on a `class` statement and an arrow function. Probably needs some modification to the build process to transform the things IE doesn't understand. But I suggest we open another ticket for that as there are many more issues with IE. (i.e. I remember seeing errors about `Object` missing `assign` from a previous test.

Comment (czosel) [reply addon/components/cf-field/input/file.js]: Oh, good to know! This sounds like something we should look at with high priority then.

**Comment by fkm** [comment]
@czosel Das Ding ist nun soweit fertig.

Die Tests sind noch nicht wirklich fertig. Aber Coverage sollte anständig sein und alles funktioniert.

**Comment by czosel** [comment]
@fkm can you bring the tests in a state that they're passing? (if needed by skipping some?)

File: addon/gql/queries/get-fileanswer-info.graphql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+query getFileAnswerInfo($id: ID!) {
+  node(id: $id) {
+    id
+
```
Comment (czosel) [inline addon/gql/queries/get-fileanswer-info.graphql]: nitpick: whitespace

Comment (fkm) [reply addon/gql/queries/get-fileanswer-info.graphql]: Whitespace? The empty line after `id`?

Comment (czosel) [reply addon/gql/queries/get-fileanswer-info.graphql]: yep - or is this something we do anywhere else already?

Comment (fkm) [reply addon/gql/queries/get-fileanswer-info.graphql]: I'll have to check. It is something I do elsewhere :smile:

Comment (fkm) [reply addon/gql/queries/get-fileanswer-info.graphql]: We don't. Change is coming with the next commit.

File: addon/components/cf-field/input/file.js [hunk 3]
```diff
@@ -6,6 +6,35 @@ import { inject as service } from "@ember/service";
...
 import layout from "../../../templates/components/cf-field/input/file";
 
+/**
+ * Promise wrapper around XHMLHttpRequest
+ *
+ * @param {File} file The file to upload.
+ * @param {String} url The MinIO uploadUrl
+ * @return {Promise<Event>} A promise resolving or rejecting with the event.
+ */
+function uploadFile(file, url) {
```
Comment (czosel) [inline addon/components/cf-field/input/file.js]: :+1: I was tempted to mention this in review as well :wink:

**Pull request merged**